test/integration/http_tls_enabled_test.lua

```diff
@@ -0,0 +1,182 @@
+            ssl_cert_file = fio.pathjoin(ssl_data_dir, 'server.crt'),
+            ssl_ca_file = fio.pathjoin(ssl_data_dir, 'ca.crt'),
+        },
+        request_opts = {
+            ssl_cert = fio.pathjoin(ssl_data_dir, 'client.crt'),
+            ssl_key = fio.pathjoin(ssl_data_dir, 'client.key'),
+        },
+    },
+    test_client_password_key_missing = {
+        ssl_opts = {
+            ssl_key_file = fio.pathjoin(ssl_data_dir, 'server.key'),
+            ssl_cert_file = fio.pathjoin(ssl_data_dir, 'server.crt'),
+            ssl_ca_file = fio.pathjoin(ssl_data_dir, 'ca.crt'),
+        },
+        request_opts = {
+            ssl_cert = fio.pathjoin(ssl_data_dir, 'client.crt'),
+            ssl_key = fio.pathjoin(ssl_data_dir, 'client.enc.key'),
+        },
+        expected_err_msg = "curl: Problem with the local SSL certificate",
+    },
+    test_ciphers_server = {
+        ssl_opts = {
+            ssl_key_file = fio.pathjoin(ssl_data_dir, 'server.key'),
+            ssl_cert_file = fio.pathjoin(ssl_data_dir, 'server.crt'),
+            ssl_ca_file = fio.pathjoin(ssl_data_dir, 'ca.crt'),
+            ssl_ciphers = "ECDHE-RSA-AES256-GCM-SHA384",
+        },
+        request_opts = {
+            ssl_cert = fio.pathjoin(ssl_data_dir, 'client.crt'),
+            ssl_key = fio.pathjoin(ssl_data_dir, 'client.key'),
+        },
+    },
+    test_invalid_key_path = {
+        ssl_opts = {
+            ssl_key_file = fio.pathjoin(ssl_data_dir, 'server.key'),
+            ssl_cert_file = fio.pathjoin(ssl_data_dir, 'server.crt'),
+            ssl_ca_file = fio.pathjoin(ssl_data_dir, 'ca.crt'),
+        },
+        request_opts = {
+            ssl_cert = fio.pathjoin(ssl_data_dir, 'client.crt'),
+            ssl_key = fio.pathjoin(ssl_data_dir, 'invalid.key'),
+        },
+        expected_err_msg = "curl: Problem with the local SSL certificate",
+    },
+    test_invalid_cert_path = {
+        ssl_opts = {
+            ssl_key_file = fio.pathjoin(ssl_data_dir, 'server.key'),
+            ssl_cert_file = fio.pathjoin(ssl_data_dir, 'server.crt'),
+            ssl_ca_file = fio.pathjoin(ssl_data_dir, 'ca.crt'),
+        },
+        request_opts = {
+            ssl_cert = fio.pathjoin(ssl_data_dir, 'invalid.crt'),
+            ssl_key = fio.pathjoin(ssl_data_dir, 'client.key'),
+        },
+        expected_err_msg = "curl: Problem with the local SSL certificate",
+    },
+}
+
+for name, tc in pairs(client_test_cases) do
+    g.before_test(name, function ()
+        g.httpd = helpers.cfgserv(tc.ssl_opts)
+        g.httpd:start()
+    end)
+
+    g.after_test(name, function ()
+        helpers.teardown(g.httpd)
+    end)
+
+    g[name] = function ()
+        local req_opts = http_server.internal.extend({
+            -- We need to provide ca_file by default due to curl uses the
+            -- system native CA store for verification.
+            -- See: https://curl.se/docs/sslcerts.html
```

Comment:
```suggestion
            -- We need to provide ca_file by default because curl uses the
            -- system native CA store for verification.
            -- See: https://curl.se/docs/sslcerts.html
```